### PR TITLE
[Form][FrameworkBundle][TwigBridge] do not depend on a fixed date in layout tests

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

By default, the `DateType` as well as the `DateTimeType` set the choices being available for the year to a range starting five years in the past. After some time, this will make tests fail when the year of the fixed date being used as the initial data is before the first year being part of the choices.

### CODE REVIEW

**Comment by chalasr** [comment]
👍  It seems to fix tests on 2.7. 
However this change concerns the Form component rather than the TwigBridge, right?

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
         "symfony/dom-crawler": "~2.0,>=2.0.5",
         "symfony/intl": "~2.3",
         "symfony/security": "~2.6",
-        "symfony/form": "~2.7,>=2.7.2",
+        "symfony/form": "~2.7.23|~2.8.16",
```
Comment (jderusse) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: `>=3.0` does not match this new pattern whereas it matched the previous one. Is it made on purpose?

File: src/Symfony/Bridge/Twig/composer.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "require-dev": {
         "symfony/asset": "~2.7",
         "symfony/finder": "~2.3",
-        "symfony/form": "~2.7.11|~2.8.4",
+        "symfony/form": "~2.7.23|~2.8.16",
```
Comment (jderusse) [inline src/Symfony/Bridge/Twig/composer.json]: Is this version bumping related to this PR?

Comment (xabbuh) [inline src/Symfony/Bridge/Twig/composer.json]: Yes, the `deps=low` tests are falling otherwise.

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.
